File: easy-projects/easy-common/src/main/java/xyz/srclab/common/base/Defaults.java
package xyz.srclab.common.base;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

public class Defaults {

    public static final Charset CHARSET = StandardCharsets.UTF_8;

    public static final Locale LOCALE = Locale.getDefault();

    public static final TimeUnit TIME_UNIT = TimeUnit.SECONDS;

    public static final int CONCURRENCY_LEVEL = 16;
}

File: srclab-common/src/main/kotlin/xyz/srclab/common/base/Defaults.kt
package xyz.srclab.common.base

import java.io.File
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets
import java.util.*
import java.util.concurrent.TimeUnit

object Defaults {

    @JvmStatic
    @get:JvmName("charset")
    val charset: Charset = StandardCharsets.UTF_8

    @JvmStatic
    val locale: Locale
        @JvmName("locale") get() {
            return Locale.getDefault()
        }

    @JvmStatic
    @get:JvmName("timeUnit")
    val timeUnit: TimeUnit = TimeUnit.SECONDS

    @JvmStatic
    @get:JvmName("concurrencyLevel")
    val concurrencyLevel: Int = 16

    @JvmStatic
    @get:JvmName("fileSeparator")
    val fileSeparator: String = File.separator

    @JvmStatic
    @get:JvmName("pathSeparator")
    val pathSeparator: String = File.pathSeparator

    @JvmStatic
    val lineSeparator: String
        @JvmName("lineSeparator") get() {
            return System.lineSeparator()
        }
}